runtime/src/bank.rs

```diff
@@ -202,6 +202,188 @@ use {
+        TransactionError::AddressLookupTableNotFound => 24,
+        TransactionError::InvalidAddressLookupTableOwner => 25,
+        TransactionError::InvalidAddressLookupTableData => 26,
+        TransactionError::InvalidAddressLookupTableIndex => 27,
+        TransactionError::InvalidRentPayingAccount => 28,
+        TransactionError::WouldExceedMaxVoteCostLimit => 29,
+        TransactionError::WouldExceedAccountDataTotalLimit => 30,
+        TransactionError::DuplicateInstruction(_) => 31,
+        TransactionError::InsufficientFundsForRent { .. } => 32,
+        TransactionError::MaxLoadedAccountsDataSizeExceeded => 33,
+        TransactionError::InvalidLoadedAccountsDataSizeLimit => 34,
+        TransactionError::ResanitizationNeeded => 35,
+        TransactionError::ProgramExecutionTemporarilyRestricted { .. } => 36,
+        TransactionError::UnbalancedTransaction => 37,
+    }
+}
+
+/// FIREDANCER: Make sure SanitizedTransaction ABI doesn't change. This
+/// doesn't really check it properly, but it's better than nothing.
+const _CHECK_ABI: [u8; 248] = [0; std::mem::size_of::<SanitizedTransaction>()];
+
+#[no_mangle]
+pub extern "C" fn fd_ext_bank_load_and_execute_txns( bank: *const std::ffi::c_void, txns: *const std::ffi::c_void, txn_count: u64, out_load_results: *mut i32, out_executing_results: *mut i32, out_executed_results: *mut i32 ) -> *mut std::ffi::c_void {
+    let txns = unsafe {
+        std::slice::from_raw_parts(txns as *const SanitizedTransaction, txn_count as usize)
+    };
+    let bank = bank as *const Bank;
+    unsafe { Arc::increment_strong_count(bank) };
+    let bank = unsafe { Arc::from_raw( bank as *const Bank ) };
+
+    let lock_results = txns.iter().map(|_| Ok(()) ).collect::<Vec<_>>();
+    let batch = TransactionBatch::new(lock_results, bank.as_ref(), Cow::Borrowed(txns));
+
+    let mut timings = ExecuteTimings::default();
+    let output = bank.load_and_execute_transactions(&batch, MAX_PROCESSING_AGE, false, false, false,
+        &mut timings, None, None);
+
+    for i in 0..txn_count {
+        match &output.loaded_transactions[i as usize].0 {
+            Ok(_) => unsafe { *out_load_results.offset(i as isize) = 0 },
+            Err(err) => unsafe { *out_load_results.offset(i as isize) = transaction_error_to_code( err ) },
+        }
+        match &output.execution_results[i as usize] {
+            TransactionExecutionResult::Executed { details, .. } => {
+                unsafe { *out_executing_results.offset(i as isize) = 0 };
+                match &details.status {
+                    Ok(_) => unsafe { *out_executed_results.offset(i as isize) = 0 },
+                    Err(err) => unsafe { *out_executed_results.offset(i as isize) = transaction_error_to_code( err ) },
+                }
+            },
+            TransactionExecutionResult::NotExecuted(err) => {
+                unsafe { *out_executing_results.offset(i as isize) = transaction_error_to_code( err ) };
```

Comment:
Having these two parallel arrays (`out_executed_results` and `out_executing_results`) that store similar error codes seems a little gross. I think it's their fault for the way they structured the errors.
What about making one just a bool if the transaction executed or not and the second can have the actual error? Are there any actual error codes that can be returned in the executing and non-executing cases? I think they should just split the error enum into two types: execution errors and sanitization (pre-execution?) errors.